src/parser/semantics.ml

```diff
@@ -290,6 +290,17 @@ let section_heading
     parsed_a_title, element
 
 
+let validate_first_page_heading status ast_element =
+  match status.parent_of_sections with
+  | Model.Paths.Identifier.Page ({file; _}, _) ->
+    begin match ast_element with
+      | Location.{value = `Heading (l, _, _); _} when l >= 0 && l < 4 -> ()
```

Comment:
I won't fight against this but I find that restriction to be quite artificial. I guess this is based on the `latex` interpretation since `4` would lead you to paragraphs and that you'd find absurd to have one `.mld` file per paragraph. 

But now if you get this interpretation out of your mind, that could simply be the valid description level of a certain subsystem in a layered architecture that you would like to document in its own file.

Reply:
Yes, your interpretation is correct. One additional supporting argument is the fact that the TOC generated by odoc should not include levels 4 and 5 – this way the content in a page would always be equivalent to something that shows up in a TOC.

I can remove this restriction, I don't feel strongly about it.

Reply:
Removed in [25ab291](https://github.com/ocaml/odoc/pull/217/commits/25ab29110979bfd637e2f5fb2bdc53e533061eac).